src/transformers/models/t5/modeling_t5.py

```diff
@@ -2121,12 +2122,15 @@ def forward(
         
         loss = None
         if labels is not None:
-            loss_fct = CrossEntropyLoss(ignore_index=-100)
+            reduction = 'mean' if sample_weights is None else 'none'
+            loss_fct = CrossEntropyLoss(ignore_index=-100, reduction=reduction)
             # move labels to correct device to enable PP
             labels = labels.to(lm_logits.device)
             loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
             # TODO(thom): Add z_loss https://github.com/tensorflow/mesh/blob/fa19d69eafc9a482aff0b59ddd96b025c0cb207d/mesh_tensorflow/layers.py#L666
-
+            if sample_weights is not None:
+                sample_weights_expanded = sample_weights.unsqueeze(1).expand(-1, labels.shape[-1]).reshape(-1)
```

Comment:
please explain the shape and content of (1)  loss, (2) labels and (3) sample_weights_expanded.

Reply:
done